pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -201,7 +201,9 @@ func (c *ClusterConfig) ValidatePrivateCluster() error {
 		if c.VPC != nil && c.VPC.ID != "" && len(c.VPC.Subnets.Private) == 0 {
 			return errors.New("vpc.subnets.private must be specified in a fully-private cluster when a pre-existing VPC is supplied")
 		}
-		if additionalEndpoints := c.PrivateCluster.AdditionalEndpointServices; len(additionalEndpoints) > 0 {
+		if additionalEndpoints := c.PrivateCluster.AdditionalEndpointServices; len(additionalEndpoints) > 0 && c.PrivateCluster.SkipEndpointCreation {
```

Comment:
can we move the variable creation above `if`, please?

Reply:
why? It's only relevant to this if statement.

Reply:
aren't you using it for the else too? anyway, I just twitch a little when I see variable assignments in `if` statements. But its upto you, not gonna block the PR for this :D

Reply:
Yeah I understand what you are saying. :D But it's quite common in Go to do this kind of inlining. Yeah the `else` is kind of part of it. :D If you really really hate it I'll change it to make you happy. :D :D

Reply:
no Gergely you do you! its perfectly fine 🎉  i've no issues :D